### PR TITLE
fix: fix the table descs of value state

### PR DESCRIPTION
**Pull request by BowenXiao1999**
## What's changed and what's your intention?

Previously I misunderstand state table. The insert was done like `insert(pk: group_key, value: agg_value)`.
But in fact, it should be done like `insert(pk: group_key, value: [group_key, agg_value])`. Otherwise the pk_indices in future refactor won't work. 

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/stream/src/executor/managed_state/aggregation/value.rs [hunk 1]
```diff
@@ -57,7 +57,8 @@ impl ManagedValueState {
                 .get_row(pk.unwrap_or(&Row(vec![])), epoch)
                 .await?;
 
-            raw_data.and_then(|row| row.values().next().cloned())
+            // According to row layout, the last field of the row is value.
+            raw_data.map(|row| row.0[row.0.len() - 1].clone())
```
Comment (BugenZhao) [inline src/stream/src/executor/managed_state/aggregation/value.rs:61]: May use `.last()` here?

File: src/stream/src/executor/aggregation/mod.rs [hunk 2]
```diff
@@ -342,6 +342,59 @@ pub fn generate_agg_schema(
...
+
+    // For max, min, the table descs should include sort key.
+    // The added columns should be (sort_key, pk from input data).
+    if (agg_call.kind == AggKind::Max || agg_call.kind == AggKind::Min) && !agg_call.append_only {
+        // Add value as part of sort key.
+        column_descs.push(ColumnDesc::unnamed(
+            ColumnId::new(column_id),
+            agg_call.return_type.clone(),
+        ));
+        column_id += 1;
```
[the remark this answers is not in the document]
Comment (BugenZhao) [reply src/stream/src/executor/aggregation/mod.rs:379]: If the memcomparable encoding is lossless for this type, then yes.

File: src/stream/src/executor/aggregation/mod.rs [hunk 3]
```diff
@@ -342,6 +342,59 @@ pub fn generate_agg_schema(
...
+    let mut column_descs = Vec::with_capacity(group_keys.len() + 1);
+    let mut column_id = 0;
+    for (idx, _) in group_keys.iter().enumerate() {
+        // The agg schema layout as [group_key schema, agg_call schema], so directly get group key
+        // return type from it for DRY.
+        column_descs.push(ColumnDesc::unnamed(
+            ColumnId::new(column_id),
+            agg_schema.fields[idx].data_type.clone(),
+        ));
+        column_id += 1;
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/mod.rs:368]: Suggest introducing a closure of `next_column_id` here. :)

File: src/stream/src/executor/aggregation/mod.rs [hunk 4]
```diff
@@ -342,6 +342,59 @@ pub fn generate_agg_schema(
...
 }
 
+/// Infer column desc for state table.
+/// The column desc layout is
+/// [ `group_key` (only for hash agg) / `sort_key` (only for simple agg) / `value`(the agg call
+/// return type)].
+/// This is the Row layout insert into state table.
+/// For different agg call, different executor (hash agg or simple agg), the layout will be
+/// different.
+pub fn generate_column_descs(
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/mod.rs:352]: Just out of curiosity, will this be generated by meta service or frontend eventually?

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: src/stream/src/executor/aggregation/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (BugenZhao) [reply src/stream/src/executor/aggregation/mod.rs:379]: - https://github.com/singularity-data/risingwave/issues/588

File: src/stream/src/executor/aggregation/mod.rs (hunk 4 above)
Comment (BowenXiao1999) [reply src/stream/src/executor/aggregation/mod.rs:352]: I think so, serveral days ago discussed with @st1page .

File: src/stream/src/executor/aggregation/mod.rs (hunk 3 above)
Comment (BowenXiao1999) [reply src/stream/src/executor/aggregation/mod.rs]: Great! The code looks more elegant

**Pull request merged**